src/main/java/recipeapplication/database/RecipeFirebaseCrud.java

```diff
@@ -22,21 +26,73 @@ public RecipeFirebaseCrud(Firestore firestore,
 
 	@Override
 	public void create(Recipe data) {
-		
-		this.firestore.collection(collection).document(data.getId()).set(data);
-		
+		System.out.println("In recipe creation section");
+
+		try {
+			if (this.firestore.collection(collection).document(data.getId()).get().get().exists() == false)  // Check if document already exists, to not allow overwriting it with create.
+				this.firestore.collection(collection).document(data.getId()).set(data);
+			else
+				throw new Exception("Failed to create, Recipe with this ID already exists!");
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		} catch (ExecutionException e) {
+			e.printStackTrace();
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+		}
 	}
 
 	@Override
-	public void read(String document) {
-		// TODO Auto-generated method stub
-		
+	public Recipe read(String document) {
+		System.out.println("In recipe reading section");
+		try {
+			DocumentSnapshot docSnap = this.firestore.collection(collection).document(document).get().get();
```

Comment:
Suggestion: since you're using this more than once, consider exporting this to a new method

Reply:
Changed to using varibales to hold the references, no need to make a new method just to use ".get().get()" on the DocumentSnapshot variable.